tedge/src/cli/connect/command.rs

```diff
@@ -331,13 +331,23 @@ fn new_bridge(
         println!("Checking if tedge-mapper is installed.\n");
 
         if which("tedge_mapper").is_err() {
-            println!("Warning: tedge_mapper is not installed. We recommend to install it.\n");
+            println!("Warning: tedge_mapper is not installed.\n");
         } else {
             service_manager
                 .start_and_enable_service(cloud.dependent_mapper_service(), std::io::stdout());
         }
     }
 
+    println!("Checking if tedge-agent and sm-c8y mapper are installed.\n");
+
+    if bridge_config.use_agent && which("tedge_agent").is_err() {
+        println!("Info: Software management is not installed. So, skipping enabling related components.\n");
+    } else {
+        service_manager.start_and_enable_service(SystemService::TEdgeSMAgent, std::io::stdout());
+        service_manager
+            .start_and_enable_service(SystemService::TEdgeSMMapperC8Y, std::io::stdout());
+    }
```

Comment:
The logic is wrong here. The agent will be started even when the agent is not used.

Using a positive test as in disconnect is less error prone.

Reply:
fixed